bob-access/src/authenticator/basic.rs

```diff
@@ -40,7 +40,14 @@ impl<Storage: UsersStorage> Basic<Storage> {
         }
         self.nodes
             .get(other.ip().as_ref()?)
-            .map(|cred| cred.username() == other.username())
+            .map(|cred| 
+                match (cred.kind(), other.kind()) {
+                    (Some(CredentialsKind::Basic { username: c_username, password: _ }),
+                    Some(CredentialsKind::Basic { username: o_username, password: _ })) => c_username == o_username,
+                    (Some(CredentialsKind::Token(c_token)), 
+                    Some(CredentialsKind::Token(o_token))) => c_token == o_token,
```

Comment:
This is incorrect. Tokens can't be compared directly. Moreover, this is a function to perform inter-node authentication: it does not use either Token or Basic authentication. Nodes use a separate authentication scheme by IP and node name. Previously it worked, because the node name was passed as username, but this was a bad approach. Now things have gotten even worse. I suggest to add a separate `CredentialsKind` (`InterNode`), that stores `node_name` inside and used here for authentication. With this approach node name should be passed in separate header named `node_name`

Reply:
Additionally, it would be good to perform inter-node authentication only for gRPC API and ignore it for REST API. This can be achieved by creating separate `check_credentials` functions for gRPC and for REST API